src/main/scala/viper/gobra/frontend/info/implementation/typing/StmtTyping.scala

```diff
@@ -103,14 +104,32 @@ trait StmtTyping extends BaseTyping { this: TypeInfoImpl =>
 
     case n@PForStmt(_, cond, _, _, _) => isExpr(cond).out ++ comparableTypes.errors(exprType(cond), BooleanT)(n)
 
-    case n@PShortForRange(exp, lefts, _) =>
-      if (lefts.forall(pointsToData)) multiAssignableTo.errors(Vector(miscType(exp)), lefts map idType)(n)
-      else error(n, s"at least one assignee in $lefts points to a type")
-
-
-    case n@PAssForRange(exp, lefts, _) =>
-      multiAssignableTo.errors(Vector(miscType(exp)), lefts map exprType)(n) ++
-        lefts.flatMap(t => addressable.errors(t)(t))
+    case _@PShortForRange(range, shorts, _, _) =>
+      if (shorts.forall(pointsToData)) {
+        val typ = miscType(range).asInstanceOf[Type.InternalTupleT]
+        if (shorts.length > 0 && shorts.length <= typ.ts.length) {
+          if (typ.ts.take(shorts.length).zip(shorts map idType).forall(assignableTo))
+            noMessages
```

Comment:
couldn't you use `multiAssignableTo` here?

Reply:
also, instead of returning all these errors explicitly, you could probably just return the errors found in multiAssignableTo (usually, the idiom that you see in code is `multiAssignableTo.errors(???, ???)(???)`)